docs/docs/misc/glossary/call_types.md

```diff
@@ -0,0 +1,177 @@
+
+It is also possible to create public functions that can _only_ be invoked by privately enqueing a call from the same contract, which can very useful to update public state after private exection (e.g. update a token's supply after privately minting). This is achieved by annotating functions with `#[aztec(internal)]`.
+
+:::warning
+Calling public functions privately leaks some privacy! The caller of the function and all arguments will be revelead, so exercise care when mixing the private and public domains.
+:::
+
+A common pattern is to enqueue public calls to check some validity condition on public state, e.g. that a deadline has not expired or that some public value is set.
+
+#include_code enqueue_check /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+To learn about alternative ways to acess public state privately, look into [Shared State](../../developers/contracts/references/storage/shared_state.md).
+
+### Public Execution
+
+Contract functions marked with `#[aztec(public)]` can only be called publicly, and are executed by the sequencer. The computation model is very similar to the EVM: all state, parameters, etc. are known to the entire network, and no data is private.
+
+Since private calls are always run in a user's device, it is not possible to perform any private execution from a public context. A reasonably good mental model for public execution is that of an EVM in which some work has already been done privately, and all that is know about it is its correctness and side-effects (new commitments and nullifiers, enqueued public calls, etc.). A reverted public execution will also revert the private side-effects.
+
+Public functions in other contracts can be called both regularly and statically, just like on the EVM.
+
+#include_code public_call /noir-projects/noir-contracts/contracts/app_subscription_contract/src/main.nr rust
+
+:::note
+This is the same function that was called by privately enqueuing a call to it! Public functions can be called either directly in a public context, or asynchronously by enqueuing in a private context.
+:::
+
+### Top-level Unconstrained
+
+Contract functions with the `unconstrained` Noir keyword are a special type of function still under development, and their semantics will likely change in the near future. They are used to perform state queries from an off-chain client, and are never included in any transaction. No guarantees are made on the correctness of the result since they rely exclusively on unconstrained oracle calls.
+
+A reasonable mental model for them is that of a `view` Solidity function that is never called in any transaction, and is only ever invoked via `eth_call`. Note that in these the caller assumes that the node is acting honestly by exectuing the true contract bytecode with correct blockchain state, the same way the Aztec version assumes the oracles are returning legitimate data.
```

Comment:
Guess similar to passing the bytecode along as an argument to the `eth_call`, e.g., it is also not deployed, but just used in local execution.

Reply:
Ah good point, though I don't think the bytecode override is part of the standard interface

Reply:
I think I'd rather not mention that here to avoid introducing even more things, specially if we'll revise this soon.